sunpy/data/data_manager/tests/test_manager.py

```diff
@@ -50,6 +50,31 @@ def test_foo():
         test_foo()
 
 
+def test_defer_download(manager, storage, downloader, data_function,tmpdir):
+    """
+    Test that files are not downloaded immediately if defer_download is True,
+    but are downloaded when get is called.
+    """
+    folder = tmpdir.strpath
+    @manager.require('test_file', [f'file://{folder}/another_file'], MOCK_HASH,defer_download=True)
+    def deferred_function():
+        pass
+
+    deferred_function()
+    assert downloader.times_called == 0
+    assert len(storage._store) == 0
+
+def test_defer_download_get(manager,storage,downloader,data_function,tmpdir):
```

Comment:
Does this need to two tests?

Reply:
not really but I thought it might be good to do separate tests for showing some difference shall I combine both of the tests into one